### PR TITLE
 enable doctests, add name argument and fix ReShuffleDataset with random state

### PR DESCRIPTION
**Pull request by boeddeker**
- enable doctests in travis
 - add name argument to new, from_dict and from_list
 - use `rng` in `__iter__` of `ReShuffleDataset`
 - fix doctests for travis

### CODE REVIEW

**Comment by boeddeker** [comment]
I added also codecov

File: lazy_dataset/core.py [hunk 1]
```diff
@@ -23,6 +27,7 @@ def new(examples: Union[list, dict], immutable_warranty: str = 'pickle'):
         examples: The data to create a new dataset from
         immutable_warranty: How to ensure immutability. Available options are
             'pickle' and 'copy'.
+        name: An optional name for the dataset. Only effects the representer.
```
Comment (tglarner) [inline lazy_dataset/core.py]: effects -> affects

File: lazy_dataset/core.py [hunk 2]
```diff
@@ -31,9 +36,9 @@ def new(examples: Union[list, dict], immutable_warranty: str = 'pickle'):
         Create a dataset from a dict:
 
         >>> import lazy_dataset
-        >>> ds = lazy_dataset.new({'a': 1, 'b': 2, 'c': 3})
+        >>> ds = lazy_dataset.new({'a': 1, 'b': 2, 'c': 3}, name='MyDataset')
         >>> ds
-          DictDataset(len=3)
+          DictDataset(name=MyDataset, len=3)
```
Comment (tglarner) [inline lazy_dataset/core.py]: I would have expected the repr output here, i.e.
DictDataset(name='MyDataset', len3)

File: lazy_dataset/core.py (hunk 1 above)
Comment (boeddeker) [reply lazy_dataset/core.py]: Done

File: lazy_dataset/core.py (hunk 2 above)
Comment (boeddeker) [reply lazy_dataset/core.py]: Done

**Pull request merged**